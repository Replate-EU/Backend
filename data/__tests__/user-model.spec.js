/* const db = require("../db-config");
const seed1 = require("../seeds/01-users").seed;
const seed2 = require("../seeds/02-volunteer_accounts").seed;
const seed3 = require("../seeds/03-business_accounts").seed;
const seed4 = require("../seeds/04-pickups").seed;
const usersDb = require("../users-model");

const newUser = {
  username: "darragh",
  password: "12345",
  user_type: "volunteer",
  contact_number: "1-234-567-78-90"
};

beforeEach(async () => {
  await seed1(db);
  await seed2(db);
  await seed3(db);
  await seed4(db);
});

describe("Users-model", () => {
  describe("getById", () => {
    it("Returns correct data for volunteer users", async () => {
      expect(await db("users")).toMatchSnapshot();
      const volunteer = await usersDb.getById(1);
      console.log(volunteer);
      expect(volunteer).toMatchSnapshot();
    });

    it("Returns correct data for business users", async () => {
      const business = await usersDb.getById(2);
      console.log(business);
      expect(business).toMatchSnapshot();
    });
  });
  describe("insert()", () => {
    it("Inserts new users correctly", async () => {
      try {
        const user = await usersDb.insert(newUser);
        expect(user).toMatchOjbect({
          ...newUser,
          id: 3,
          account_details: undefined
        });
      } catch (error) {
        console.log(error);
      }
    });
  });
  describe("getByUsername()", () => {
    it("Returns correct data", async () => {
      const user = await usersDb.getByUsername("imhelping");
      expect(user).toMatchSnapshot();
    });
  });
  describe("update()", () => {
    it("Returns correct data", async () => {
      const user = await usersDb.update(newUser, 1);
      expect(user).toMatchSnapshot();
    });
  });
  describe("remove()", () => {
    it("Deletes correct user", async () => {
      const userDeleted = await usersDb.remove(1);
      expect(userDeleted).toBe(1);
      const checkForDeleted = await db("users")
        .where("id", 1)
        .first();
      expect(checkForDeleted).not.toBeDefined();
    });
  });
});
 */