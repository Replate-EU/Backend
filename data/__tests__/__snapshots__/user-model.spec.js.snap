// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users-model getById Returns correct data for business users 1`] = `
Object {
  "account_details": Object {
    "address": "1 Doughnut St.",
    "id": 1,
    "name": "Local Doughnuts",
    "user_id": 2,
  },
  "contact_number": "+4-408-048-99-16",
  "id": 2,
  "user_type": "business",
  "username": "fooddotcom",
}
`;

exports[`Users-model getById Returns correct data for volunteer users 1`] = `
Array [
  Object {
    "contact_number": "+4-408-048-99-16",
    "id": 1,
    "password": "$2a$12$ZQwXBTq7UMgmugpy5zz9SOdG4JvEa3Bj5MofQl9fIMFb1wTSGU9.C",
    "user_type": "volunteer",
    "username": "imhelping",
  },
  Object {
    "contact_number": "+4-408-048-99-16",
    "id": 2,
    "password": "$2a$12$ZQwXBTq7UMgmugpy5zz9SOdG4JvEa3Bj5MofQl9fIMFb1wTSGU9.C",
    "user_type": "business",
    "username": "fooddotcom",
  },
]
`;

exports[`Users-model getById Returns correct data for volunteer users 2`] = `
Object {
  "account_details": Object {
    "id": 1,
    "name": "Sunny McGoodperson",
    "user_id": 1,
  },
  "contact_number": "+4-408-048-99-16",
  "id": 1,
  "user_type": "volunteer",
  "username": "imhelping",
}
`;

exports[`Users-model getByUsername() Returns correct data 1`] = `
Object {
  "contact_number": "+4-408-048-99-16",
  "id": 1,
  "password": "$2a$12$ZQwXBTq7UMgmugpy5zz9SOdG4JvEa3Bj5MofQl9fIMFb1wTSGU9.C",
  "user_type": "volunteer",
  "username": "imhelping",
}
`;

exports[`Users-model update() Returns correct data 1`] = `
Object {
  "account_details": Object {
    "id": 1,
    "name": "Sunny McGoodperson",
    "user_id": 1,
  },
  "contact_number": "1-234-567-78-90",
  "id": 1,
  "user_type": "volunteer",
  "username": "darragh",
}
`;
